File: 2020/erik/Day06/src/Day06.jl
module Day06

function readdata(path)
    open(path) do file
        data = []
        group = []
        for line in eachline(file)
            if length(line) == 0
                push!(data, group)
                group = []
            else
                push!(group, Set(line))
            end
        end
        push!(data, group)
        data
    end
end

function part1(data)
    mapreduce(length, +, map(group -> reduce(union, group), data))
end

function part2(data)
    mapreduce(length, +, map(group -> reduce(intersect, group), data))
end

function main()
    data = readdata("input/input06.txt")
    println("> Part one: ", part1(data))
    println("> Part two: ", part2(data))
end

end # module

File: archive/2021/andehen/day06/day06.py
S = [0] * 9
with open("input.txt") as f:
    for f in f.readline().split(","):
        S[int(f)] += 1

s = list(S)
for d in range(1, 257):
    n = s[0]
    s = s[1:] + [n]
    s[6] += n
    if d == 80:
        print(sum(s))

print(sum(s))
